pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -430,13 +446,44 @@ impl<T: Config> Pallet<T> {
 		delegator: T::AccountId,
 		delegation_count: u32,
 	) -> DispatchResultWithPostInfo {
-		let state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+		let mut state = <DelegatorState<T>>::get(&delegator).ok_or(<Error<T>>::DelegatorDNE)?;
+
 		ensure!(
 			delegation_count >= (state.delegations.0.len() as u32),
 			Error::<T>::TooLowDelegationCountToLeaveDelegators
 		);
 		let now = <Round<T>>::get().current;
 
+		// backwards compatible handling for DelegatorStatus::Leaving
+		#[allow(deprecated)]
```

Comment:
We are sure this does not impact benchmarking right? I mean that this code path has less weight than the other path

Reply:
This would be hard to benchmark, since we have two conditional paths now - we can benchmark both but merging them will be hard. That being said the [weights](https://github.com/PureStake/moonbeam/pull/1515/files#diff-8105d5e5ee125a16967fcea164e5ec560a2c7347252405f26e9a0cdf4b05ad5a) of the deprecated extrinsic are quite comparable, if not lower than what we have right now. I'd be fine with the current weights.